harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -513,17 +497,39 @@ def _stop_requested(self) -> None:
         if self.context.get_stop_requested():
             raise ShouldExit()
 
-    def _report_searcher_progress(
-        self, op: core.SearcherOperation, unit: Optional[core.Unit]
-    ) -> None:
+    def _report_training_progress(self) -> None:
         assert self.state
-        if unit == core.Unit.BATCHES:
-            op.report_progress(self.state.batches_trained)
-        elif unit == core.Unit.RECORDS:
-            assert self.global_batch_size, "global_batch_size must be specified for RECORDS"
-            op.report_progress(self.global_batch_size * self.state.batches_trained)
-        elif unit == core.Unit.EPOCHS:
-            op.report_progress(self.state.epochs_trained)
+        if isinstance(self.max_length, Batch):
+            progress = self.state.batches_trained / self.max_length.value
+        elif isinstance(self.max_length, Epoch):
+            progress = self.state.epochs_trained / self.max_length.value
+        else:
+            raise ValueError(f"unexpected train unit type {type(self.max_length)}")
+
+        self.core_context.train.report_progress(progress=progress)
+
+    def _report_validation_metrics(self, metrics: Dict[str, Any]) -> None:
+        assert self.state
+        if isinstance(self.max_length, Batch):
+            if "batch" in metrics:
+                logger.warning(
+                    f"not overriding existing `batch: {metrics['batch']}` in validation metrics"
+                )
+            else:
+                metrics.update({"batch": self.state.batches_trained})
+        elif isinstance(self.max_length, Epoch):
+            if "epoch" in metrics:
+                logger.warning(
+                    f"not overriding existing `epoch: {metrics['epoch']}` in validation metrics"
+                )
+            else:
+                metrics.update({"epoch": self.state.epochs_trained})
+        else:
+            raise ValueError(f"unexpected train unit type {type(self.max_length)}")
```

Comment:
I don't care for this trying to guess which special metric to add.

I think, either we add `steps_completed` and `epoch` to validation metrics always, or we do it never.  I don't see why max_length should affect that.